orttraining/orttraining/core/optimizer/memory_optimizer.h

```diff
@@ -2,10 +2,14 @@
 // Licensed under the MIT License.
 
 #pragma once
-#include <charconv>
+
 #include "core/common/inlined_containers.h"
 #include "core/common/string_utils.h"
 #include "core/optimizer/graph_transformer.h"
+#include "orttraining/core/optimizer/memory_optimizer/common.h"
+#include "orttraining/core/optimizer/memory_optimizer/optimization_planner.h"
+#include "orttraining/core/optimizer/memory_optimizer/recompute_analysis.h"
+#include "orttraining/core/optimizer/memory_optimizer/memory_insight.h"
```

Comment:
could we put an algo description here, talking about the major idea of the optimizer, such as how to  find recomoutable subgraphs, how it workes

Reply:
Added